### DISCUSSION TITLE
Make "master" the first branch which shows up under the branch dropdown

### DISCUSSION THREAD

**Discussion by AkshayBanifatsyevichChugh** [gitextensions/gitextensions#9877] [Q&A] [answered]
### Feature description

Currently when you press the dropdown next to the current branch on the main ui screen it lists branches in alphabetical order descending. Change this so that it either displays master as the first branch or implements the Most recently used ordering. Currently if repository has a lot of branches it's a pain to have to look for the master branch.
![image](https://user-images.githubusercontent.com/46527900/155513093-12b5f670-4b3f-4f24-aaf6-c2f24294c787.png)

### Environment

- Git Extensions 3.5.4.12724
- Build 65f01f39982ccb121c4df7d4b00b506939553958
- Git 2.34.1.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4084.0
- DPI 96dpi (no scaling)

**Comment by pmiossec** [comment, marked as answer]
> Currently when you press the dropdown next to the current branch on the main ui screen it lists branches in alphabetical order descending.

In fact, that depends of your preferences. 

>Change this so that it either displays master as the first branch

That makes little sense to hardcode a branch name especially now that git allows to have a different default branch. Maybe gitextensions could look if the git setting specify a default branch is set and put that one first but definitely not master... 

> or implements the Most recently used ordering.

Have a look at the settings, that's already possible.

**Reply by RussKie** [reply]
![image](https://user-images.githubusercontent.com/4403806/155831492-856d8bf7-4ec8-49bc-95ef-75353b8c2577.png)

**Reply by pmiossec** [reply]
In fact, determining the default branch name is more difficult than expected:
https://github.blog/2020-07-27-highlights-from-git-2-28/#introducing-init-defaultbranch

**Comment by RussKie** [comment]
The default branch is a _convention_, generally agreed upon in a team:
master, main, dev, etc.

**Comment by gerhardol** [comment]
There has been discussions to have a configuration (I suggest a regex) list (repo, global) to affect sorting of branches in sidepanel, order in grid, commit, dropdowns etc that is always displayed first in these lists.

**Comment by Temppus** [comment]
I find alphabetical order of branches in dropdown pretty confusing and I am struggling 😠 to find branch that I am interested in because in 90%  of cases I want to switch on branch which I recently worked on (ordered by time).

**Reply by gerhardol** [reply]
> I find alphabetical order of branches in dropdown pretty confusing and I am struggling 😠 to find branch that I am interested in because in 90% of cases I want to switch on branch which I recently worked on (ordered by time).

![image](https://user-images.githubusercontent.com/6248932/175014472-c15ec074-d2e9-499f-abbe-37675e2e9d45.png)
